### PR TITLE
11436 + 17937 - Framework for AGM Extension and AGM Location Change filings

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#11436 and bcgov/entity#17937

Please note that the code duplication is inevitable until we refactor all local filings into a common template + slots (which will be a lot of work and is not planned atm).

*Description of changes:*

Commit 1:
    - app version = 6.9.0
    - imported some updated shared components
    - added routes
    - added menus to Entity Menu
    - added allowable actions enum keys
    - added routes enum keys
    - added allowable actions mixin cases
    - added default FFs
    - created initial AGM Extension view
    - created initial AGM Location Chg view
    - added some Entity Menu tests

Commit 2:
    - added routes to "is local filing"
    - added override to Allowable Actions mixin (TEMPORARY)
    - updated initial AGM Extension view
    - updated initial AGM Location Chg view
    - fixed a lint warning

Commit 3:
    - added shared Expandable Help component
    - fixed FF in template not working
    - added business config objects
    - fixed section rows
    - added initial help content
    - created initial unit tests for views
    - fixed unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-547-f6vodu71.web.app

File: package-lock.json [hunk 1]
```diff
@@ -1,24 +1,25 @@
 {
   "name": "business-filings-ui",
-  "version": "6.8.1",
+  "version": "6.9.0",
```
Comment (severinbeauvais) [inline package-lock.json:3]: I increased the minor version number as we are adding something significant.

File: src/components/EntityInfo/EntityMenu.vue [hunk 2]
```diff
@@ -163,6 +163,48 @@
...
             </v-tooltip>
+
+            <!-- Request AGM Extension -->
+            <v-tooltip
+              right
+              content-class="right-tooltip"
+            >
+              <template #activator="{ on }">
+                <v-list-item
+                  v-if="enableAgmExtension"
```
Comment (severinbeauvais) [inline src/components/EntityInfo/EntityMenu.vue:174]: Only show this menu item if the feature is enabled via FF.

Oddly, this didn't work when I put the FF check here directly, so I put it in a getter (computed) instead.

File: src/components/EntityInfo/EntityMenu.vue [hunk 3]
```diff
@@ -163,6 +163,48 @@
...
+                  id="agm-ext-list-item"
+                  :disabled="!isAllowed(AllowableActions.AGM_EXTENSION)"
```
Comment (severinbeauvais) [inline src/components/EntityInfo/EntityMenu.vue:176]: Show as disabled (grayed out) if this action is not allowable (ie, Legal API checks some things for this entity such as state, etc).

File: src/components/common/FileUploadPdf.vue [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <template>
   <v-file-input
-    class="file-upload-pdf"
     :key="count"
+    class="file-upload-pdf"
```
Comment (severinbeauvais) [inline src/components/common/FileUploadPdf.vue:4]: Fixed lint issue.

File: src/mixins/allowable-actions-mixin.ts [hunk 5]
```diff
@@ -39,6 +39,16 @@ export default class AllowableActionsMixin extends Vue {
         return this.isAllowedFiling(FilingTypes.ANNUAL_REPORT)
       }
 
+      case AllowableActions.AGM_EXTENSION: {
+        return true // *** FOR DEBUGGING ONLY
```
Comment (severinbeauvais) [inline src/mixins/allowable-actions-mixin.ts:43]: Will be updated in bcgov/entity#18024.

File: src/mixins/allowable-actions-mixin.ts [hunk 6]
```diff
@@ -39,6 +39,16 @@ export default class AllowableActionsMixin extends Vue {
...
+        // return this.isAllowedFiling(FilingTypes.AGM_EXTENSION)
+      }
+
+      case AllowableActions.AGM_LOCATION_CHG: {
+        return true // *** FOR DEBUGGING ONLY
```
Comment (severinbeauvais) [inline src/mixins/allowable-actions-mixin.ts:48]: Will be updated in bcgov/entity#18027.

File: tests/unit/AgmExtension.spec.ts [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+
+Vue.use(Vuetify)
+
+const localVue = createLocalVue()
+localVue.use(VueRouter)
+setActivePinia(createPinia())
+const businessStore = useBusinessStore()
+const rootStore = useRootStore()
+
+describe.skip('AGM Extension view', () => {
```
Comment (severinbeauvais) [inline tests/unit/AgmExtension.spec.ts:33]: Needs to be implemented.

File: tests/unit/AgmLocationChg.spec.ts [hunk 8]
```diff
@@ -0,0 +1,325 @@
...
+
+Vue.use(Vuetify)
+
+const localVue = createLocalVue()
+localVue.use(VueRouter)
+setActivePinia(createPinia())
+const businessStore = useBusinessStore()
+const rootStore = useRootStore()
+
+describe.skip('AGM Location Chg view', () => {
```
Comment (severinbeauvais) [inline tests/unit/AgmLocationChg.spec.ts:33]: Needs to be implemented.

File: tests/unit/EntityMenu.spec.ts [hunk 9]
```diff
@@ -456,22 +457,70 @@ describe('Entity Menu - More actions click tests', () => {
...
+})
+
+describe('Entity Menu - Request AGM Extension click tests', () => {
   beforeAll(() => {
-    // mock the window.location.assign function
-    delete window.location
-    window.location = { assign: vi.fn() } as any
+    // override feature flag
+    vi.spyOn(utils, 'GetFeatureFlag').mockImplementation(
+      (flag) => (flag === 'enable-agm-extension'))
```
Comment (severinbeauvais) [inline tests/unit/EntityMenu.spec.ts:485]: Here is one way to mock the FF. See also line 490.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-547-f6vodu71.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-547-f6vodu71.web.app

File: src/components/EntityInfo/EntityMenu.vue (hunk 2 above)
Comment (ketaki-deodhar) [reply src/components/EntityInfo/EntityMenu.vue:174]: I have experienced something similar (do not remember exactly which code base) but works in a getter

**Review by ketaki-deodhar — APPROVED** [review]
Looks good to me!

File: src/components/EntityInfo/EntityMenu.vue (hunk 2 above)
Comment (severinbeauvais) [reply src/components/EntityInfo/EntityMenu.vue:174]: Actually, this worked when running locally, but not in the unit tests 🤷‍♂️

File: tests/unit/EntityMenu.spec.ts (hunk 9 above)
Comment (severinbeauvais) [reply tests/unit/EntityMenu.spec.ts:485]: PS - I could have simply mocked the return value, but this is more explicit that it's only True for THIS flag.

**Review by leodube-aot — APPROVED** [review]

**Review by chenhongjing — APPROVED** [review]
LGTM 🐱

**Comment by tshyun24** [comment]
Looks good to me also👏

**Review by tshyun24 — APPROVED** [review]

**Review by PaulGarewal — APPROVED** [review]
LGTM 👍

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-547-f6vodu71.web.app

**Pull request merged**